### ISSUE TITLE
%pyproject_wheel doesn't work without %python_subpackages

### ISSUE THREAD

**Issue by mcepl** [openSUSE/python-rpm-macros#146]
When I simplify the SPEC file for python-dnspython to this non-multi-version form the build fails:
```rpm
Name:           python-dnspython
Version:        2.2.1
Release:        0
Summary:        A DNS toolkit for Python
License:        ISC
Group:          Development/Languages/Python
URL:            https://github.com/rthalley/dnspython
Source:         https://files.pythonhosted.org/packages/source/d/dnspython/dnspython-%{version}.tar.gz
BuildRequires:  python3-base >= 3.6
BuildRequires:  python3-pip
BuildRequires:  python3-poetry-core
BuildRequires:  fdupes
BuildRequires:  python-rpm-macros
BuildRequires:  python3-typing
# doh:
BuildRequires:  python3-httpx
BuildRequires:  python3-h2
BuildRequires:  python3-requests-toolbelt
BuildRequires:  python3-requests
# idna
BuildRequires:  python3-idna
# dnssec
BuildRequires:  python3-cryptography
# trio
BuildRequires:  python3-trio >= 0.14.0
# curio
BuildRequires:  python3-sniffio >= 1.1
BuildRequires:  python3-curio >= 1.2
BuildRequires:  python3-pytest
BuildRequires:  netcfg
# Requires despite optional: see description
# doh
Requires:       python3-requests
Requires:       python3-httpx
Requires:       python3-requests-toolbelt
# idna
Requires:       python3-idna >= 2.1
# dnssec
Requires:       python3-cryptography
# trio
Suggests:       python3-trio >= 0.14.0
# curio
Suggests:       python3-sniffio >= 1.1
Suggests:       python3-curio >= 1.2
# HTTP/2 support in httpx
Recommends:     python3-h2
BuildArch:      noarch

%description
dnspython is a DNS toolkit for Python. It supports almost all
record types. It can be used for queries, zone transfers, and
dynamic updates. It supports TSIG authenticated messages and EDNS0.

dnspython provides both high and low level access to DNS. The high
level classes perform queries for data of a given name, type, and
class, and return an answer set. The low level classes allow direct
manipulation of DNS zones, messages, names, and records.

The package requires dependencies necessary for these optional features:
- DNS over HTTPS (doh)
- IDNA
- DNSSEC
and suggest dependencies necessary for these optional features:
- trio
- curio
This optional feature is not available due to missing dependencies:
- wmi

%prep
%autosetup -p1 -n dnspython-%{version}
chmod -x examples/*
# https://github.com/rthalley/dnspython/pull/755
chmod -x dns/win32util.py

%build
%pyproject_wheel

%install
%pyproject_install
%fdupes %{buildroot}%{python3_sitelib}/

%check
%pytest

%files
%license LICENSE
%doc README.md examples/
%{python3_sitelib}/dns
%{python3_sitelib}/dnspython-%{version}*-info

%changelog
```

The build fails with this error in the `%build` section:

```
[    3s] Executing(%build): /usr/bin/bash -e /var/tmp/rpm-tmp.w0NQuu
[    3s] + umask 022
[    3s] + cd /home/abuild/rpmbuild/BUILD
[    3s] + /usr/bin/rm -rf /home/abuild/rpmbuild/BUILDROOT/python-dnspython-2.2.1-0.x86_64
[    3s] ++ dirname /home/abuild/rpmbuild/BUILDROOT/python-dnspython-2.2.1-0.x86_64
[    3s] + /usr/bin/mkdir -p /home/abuild/rpmbuild/BUILDROOT
[    3s] + /usr/bin/mkdir /home/abuild/rpmbuild/BUILDROOT/python-dnspython-2.2.1-0.x86_64
[    3s] + cd dnspython-2.2.1
[    3s] ++ '[' -f _current_flavor ']'
[    3s] ++ true
[    3s] + last_flavor=
[    3s] + '[' -z '' ']'
[    3s] + last_flavor=tmp
[    3s] + '[' tmp '!=' python38 ']'
[    3s] + '[' -d build ']'
[    3s] + '[' -d _build.python38 ']'
[    3s] + echo python38
[    3s] + python_flavor=python38
[    3s] + myargs=
[    3s] + '[' -f './dist/*-none-any.whl' ']'
[    3s] + /usr/bin/python3.8 -mpip wheel --verbose --progress-bar off --disable-pip-version-check --use-pep517 --no-build-isolation --no-deps --wheel-dir ./build .
[    3s] /var/tmp/rpm-tmp.w0NQuu: line 50: /usr/bin/python3.8: No such file or directory
[    3s] error: Bad exit status from /var/tmp/rpm-tmp.w0NQuu (%build)
```

Apparently, `%pyproject_wheel` didn’t recognize that it is not running in the multi-version mode and it tried to work with multiple Pythons anyway. I had to redefine `%pythons` just to `python310` to make this SPEC file working correctly.
Would it be possible to fix `%pyproject_wheel` and `%pyproject_install` to use just the primary Python interpreter for this situation, please?

**Comment by bnavigator** [comment]
This is by design. `%python_subpackages` and flavor expansion in `%build`, `%install`, and `%check` are independent of each other.

`%python_setup` and `%python_install` don't work either, do they?

Use `%python3_pyproject_wheel` (available since python-rpm-macros 20220912) as you would do `%python3_setup` in a non-multi-version setup.

**Comment by mcepl** [comment]
Kewl. Thank you.
